feat(did-key): Add EBSI Natural Person Format

ianhamilton87 opened the pull request:
Enhance did:key resolution to also be able to resolve 

See: https://hub.ebsi.eu/vc-framework/did/natural-person
(But know that EBSI likes to move around their documentation and break links from time to time, so if this doesn't resolve in the future, search for `EBSI natural person did:key`)

implement from_multicodec for jwk_jcs_pub multicodec.
 Also needed to promote serde_json from a dev-dependency to a regular dependency.

Relevant issues that are related to this PR:
https://github.com/w3c-ccg/did-method-key/issues/63
https://github.com/spruceid/ssi/issues/536

Commit:
update from main

Commit:
DID-876 Did:key support for resolving EBSI v2

 implement from_multicodec for jwk_jcs_pub multicodec.
 Also needed to promote serde_json from a dev-dependency to a regular dependency.

Commit:
DID-876 Did:key support for resolving EBSI v2

 implement from_multicodec for jwk_jcs_pub multicodec.
 Also needed to promote serde_json from a dev-dependency to a regular dependency.

Commit:
feat(did-key): Add EBSI Natural Person Format

Fixed link to EBSI docs

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/spruceid/ssi?pullRequest=545) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/spruceid/ssi?pullRequest=545) before we can accept your contribution.<br/>**4** out of **5** committers have signed the CLA.<br/><br/>:white_check_mark: mrappard<br/>:white_check_mark: timothee-haudebourg<br/>:white_check_mark: rschulman<br/>:white_check_mark: ivan770<br/>:x: Ian Hamilton<br/><hr/>**Ian Hamilton** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/spruceid/ssi?pullRequest=545) it.</sub>

Commit:
feat(did-key): Add EBSI Natural Person Format

Update rust version due to failure in `Test WASM compilation`

ianhamilton87 commented on .github/workflows/build.yml:
```diff
@@ -15,68 +15,68 @@ jobs:
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: "1.73.0"
```
The one and only true change in this file - upped the toolchain to 1.73.0

sbihel commented:
Thank you for the PR, for the time being we'll mark it as blocked by #508.

timothee-haudebourg committed:
Complete refactor (#508)

Spanning VCs, DIDs, VMs, and more.

timothee-haudebourg committed:
Token Status List & Bitstring Status List v1.0 (#551)

* Add `ssi-status` library implementing IETF and W3C bitvec status lists.

timothee-haudebourg committed:
One step signature and verification (#552)

* Remove the `Verifiable` type.

* Simplify signature/verification environment using latest `json-ld`.

timothee-haudebourg committed:
VCDM v2.0 (#549)

* Data Model v2.0 traits.
* VCDM v2.0 (JSON) syntax types.
* Add `DataIntegrityDocument` type.
* Add documentation, defaults & re-exports.

mrappard committed:
Relax DateTimeEnvironment impl requirements (#557)

Co-authored-by: ivan770 <ivan@ivan770.me>

sbihel committed:
Allow both v1 and v2 VCs in JsonCredentialOrJws (#554)

Also:
* Make present example compile, and add the scenarios as tests

sbihel committed:
Add `Ed25519Signature2020` context when missing (#560)

Also:
- add end-to-end test for signing a `Ed25519Signature2020` VCDM v1 credential
- tweak `AddProofContext` to append contexts instead of replacing

timothee-haudebourg committed:
Add accessors to the VCDM `Context` type. (#561)

Add `Context::contains` method.
Add `Context::iter` method.
Add `Context::insert` method.
Impl `IntoIterator` for `&Context`.
Impl `IntoIterator`for `Context`.
Impl `Extend` for `Context`.
Impl `From<Context>` for `json_ld::Context`.
Impl `TryFrom<json_ld::Context>` for `Context`.

sbihel committed:
Improve InvalidTypeName error (#563)

sbihel committed:
Fix evidences typo (#564)

Evidence is uncountable

timothee-haudebourg committed:
Skip serializing missing id. (#565)

timothee-haudebourg committed:
Reject `null` identifiers in VCDM. (#567)

ivan770 committed:
Prefer OUT_DIR for outputting the multicodec table (#558)

timothee-haudebourg committed:
Add convenient constructors for status lists. (#566)

timothee-haudebourg committed:
Verifier simplification. (#559)

* Merge `context` and `verifier`.
* Rename `with_default_options` into `into_vm_resolver`.
* Rename `Validate` into `ValidateClaims` trait.
* Rename `*Environment` traits into `*Provider` traits.
* Add `ResolverProvider` trait.
* Rename `Verifier` type into `VerificationParameters`.
* Add dedicated `verify` methods for secured claims types (`CompactJWS`, `DecodedJWS` and `DataIntegrity`).
* Optional date time parameter in `VerificationParameters`.
* Add documentation.

timothee-haudebourg committed:
Fix `token_status_list::StatusSize` deserializer. (#568)

sbihel committed:
Add data integrity v2 context (#569)

timothee-haudebourg committed:
Add `NonEmptyObject` type. (#570)

Set default VCDM v1 credential subject to `Object`.
Set default VCDM v2 credential subject to `NonEmptyObject`.

timothee-haudebourg committed:
Re-export `ssi-status` as `ssi::status`. (#571)

* Move and re-export `ssi-status`.
* Make the `ssi-security::multibase` module public.

timothee-haudebourg committed:
`ecdsa-rdfc-2019` cryptosuite. (#555)

* Add `ecdsa-rdfc-2019` implementation.
* Update `credentials/v2` context.
* Fix tests.
* Remove debug print.
* Add e2e tests for ecdsa
* Fix `Multikey` to accept any algorithm, not just `EdDSA`.
* Fix features.
Co-authored-by: Simon Bihel <simon.bihel@spruceid.com>

timothee-haudebourg committed:
Rename `bitstream_status_list` module into `bitstring_status_list`. (#573)

* Remove `BitstringStatusListType`.
* Remove `BitstringStatusListEntryType`.
* Make `BitstringStatusListEntry` fields public.
* Add constructor for `BitstringStatusListEntry`.
* Add constructors and accessors for `bitstring_status_list::StatusPurpose`.
* Impl `Display` for `bitstring_status_list::StatusPurpose`.

timothee-haudebourg committed:
Upgrade `json-ld` to version 0.20 (#574)

* Add a few JSON-LD expansion tests.

sbihel committed:
Enforce non-empty subject for VCDM v2 (#572)

---------
Co-authored-by: Timothée Haudebourg <timothee.haudebourg@spruceid.com>

timothee-haudebourg committed:
Data Integrity `bbs-2023` Cryptosuite (#550)

* Add bbs2023 suite.
* Factorize tests, ensure JSON canonicalization.
* Upgrade `uuid`to 1.9
* Lazy `Multikey` decoding.
* Unification of the signature options and verification options interface.
* Add `SelectiveCryptographicSuite` trait, providing a nice interface to SD suites.
* Add `DataIntegrity::select` method.
* Add `Bbs2023` variant to `AnySuite`.
* Upgrade `json-ld` to version 0.21.1.
* Remove feature gates on `Multikey`.

---------

Co-authored-by: Simon Bihel <simon.bihel@spruceid.com>

sbihel committed:
Modify tests for unmapped types (#575)

sbihel committed:
Make DataIntegrityProof created field optional (#576)

timothee-haudebourg committed:
Fully specializable credentials. (#577)

timothee-haudebourg committed:
Add support for `Multikey` in `did:jwk`. (#578)

timothee-haudebourg committed:
Insert required types in `vc::syntax::Types::default`. (#579)

timothee-haudebourg committed:
Rename `eddsa-2022` into `eddsa-rdfc-2022`. (#581)

* Bring `eddsa-2022` back for back-compatibility.

rschulman committed:
Fix typo in README (#583)

timothee-haudebourg committed:
`ecdsa-sd-2023` cryptosuite. (#582)

* Fix sd dataset canonicalization

sbihel committed:
Fix DataIntegrity `created` type (#584)

timothee-haudebourg committed:
Add JOSE VC/VP signature format. (#586)

sbihel committed:
Bump ssi-contexts version

sbihel committed:
Remove unused circular dev dependency

sbihel committed:
Fix rustdocs broken links and add check in CI (#590)

Close #588
Close #589

Commit:
Merge remote-tracking branch 'origin/did-876-did-key-ebsi-format' into did-876-did-key-ebsi-format

# Conflicts:
#	.github/workflows/build.yml
#	crates/jwk/src/lib.rs
#	did-key/src/lib.rs
#	ssi-jwk/src/multicodec.rs
